pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1295,7 +1314,50 @@ func (cf *changeFrontier) Start(ctx context.Context) {
 	}()
 }
 
+func (cf *changeFrontier) runBillingMetricReporting(ctx context.Context) {
+	if cf.spec.JobID == 0 { // don't bill for core (non-enterprise) changefeeds
+		return
+	}
+
+	// Deregister the billing metric when the goroutine exits so we don't overbill.
+	defer cf.metrics.PerFeedAggMetrics.DeregisterJob(cf.spec.JobID)
+
+	var t timeutil.Timer
+	defer t.Stop()
+	for ctx.Err() == nil {
+		ivl := jitter(changefeedbase.BillingMetricsReportingInterval.Get(&cf.flowCtx.Cfg.Settings.SV))
```

Comment:
Q: sleeping at the end of the loop instead of the beginning is convenient for tests, but could it cause issues in degenerate situations?